packages/core/src/getters/discriminators.ts

```diff
@@ -16,7 +17,9 @@ export const resolveDiscriminators = (
         let subTypeSchema;
 
         try {
-          const { name } = getRefInfo(mappingValue, context);
+          const { originalName } = getRefInfo(mappingValue, context);
+          // name from getRefInfo may contain a suffix, which we don't want
+          const name = pascal(originalName)
```

Comment:
`name` from `getRefInfo ` includes the suffix (when configured) but we need to search in `transformedSchemas` without the suffix.
I didn't find a good way of retrieving the name without the suffix, so this is my suggestion, not sure if there's a better alternative.